### PR TITLE
Add support for "-1" values to disable checks

### PR DESCRIPTION
**Pull request by wbinnssmith**
This brings support for using negative values to have the plugin ignore rules. It looks like [Automattic forked this](https://www.npmjs.com/package/esformatter-collapse-objects-a8c) just to ignore things like maxDepth (which I admit might be a bit too much magic of a rule).

This also clarifies many things in the README, including this new option, and ideally gives a better introduction to the plugin. Examples are coming to the README soon...

### CODE REVIEW

**Comment by wbinnssmith** [comment]
cc @millermedeiros @kewah more goodies to run by you :smile: 

also cc @blowery hopefully this addresses your need for the fork. Let me know if there's anything still outstanding that's preventing you from using this upstream plugin.

File: index.js [hunk 1]
```diff
@@ -42,30 +42,28 @@ function transform(node) {
 
   // It collapses objects that are short enough
   // 0 indicates measurement failed, ignore
-  if ('maxLineLength' in nodeOptions) {
```
Comment (wbinnssmith) [inline index.js]: pretty sure with the deep-defaults module and this being set, this is unnecessary unless somebody deliberately `delete`s these keys from the config...

File: README.md [hunk 2]
```diff
@@ -18,33 +25,18 @@ and something like this to your esformatter config file:
...
+
+This plugin works by relying on esformatter expanding the relevant expressions,
+and conditionally collapsing them back down to a single line. Therefore, you
+need to have esformatter expand them in the first place (esformatter defaults to
+expanding object literals, but not array literals).
 
-In order to collapse Array literals, you need to have esformatter expand them in
-the first place. Add the following to your esformatter config when collapsing
-Arrays:
+Add the following to your esformatter config when collapsing arrays:
```
Comment (millermedeiros) [inline README.md]: Instead of adding this note why not override the settings during `setOptions` automatically?

File: README.md [hunk 3]
```diff
@@ -58,27 +50,21 @@ Arrays:
...
-ArrayExpression) to their respective options, just like indentation in
-esformatter.
-
-You can also avoid collapsing literals under certain conditions
-like a maximum number of keys, or when they contain other nodes like
-FunctionExpression.
+Since expressions were collapsed, you might be surprised that some appear as
+`{a:b,c:d}`. This is because esformatter defaults are tuned toward expanded
+expressions. Try merging the following into your config for whitespace before
+each property in an object:
```
Comment (millermedeiros) [inline README.md]: Maybe it's time to change the esformatter defaults!

File: README.md (hunk 2 above)
Comment (millermedeiros) [inline README.md]: I know this would be separate pull request tho

Comment (wbinnssmith) [inline README.md]: Hmm.. think it's okay to opt users into this behavior? Maybe they want this for objects, but would prefer arrays are always left alone. Not sure, but I like the idea of not pushing this implicit dependency onto the user.

**Comment by blowery** [comment]
> It looks like Automattic forked this just to ignore things like maxDepth (which I admit might be a bit too much magic of a rule).

It was two things really, ignoring maxDepth and still formatting objects that were part of an array, or a property of an object.

**Comment by blowery** [comment]
I'll give it a shot and see if it passes our tests. Thanks for noticing our fork!
